### PR TITLE
Migrate Botswana holidays to holiday groups

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Migrate Botswana holidays to `holiday_groups` standard.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/botswana.py [hunk 1]
```diff
@@ -30,58 +29,61 @@ class Botswana(HolidayBase):
...
-                    self[
-                        hol_date + td(days=+2 if hol_name2 else +1)
-                    ] = f"{hol_name1} (Observed)"
+    def __init__(self, *args, **kwargs) -> None:
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _add_observed(self, dt: date) -> None:
+        if self.observed and self._is_sunday(dt):
```
Comment (arkid15r) [inline holidays/countries/botswana.py:38]: The 1995 check goes away?

File: holidays/countries/botswana.py [hunk 2]
```diff
@@ -30,58 +29,61 @@ class Botswana(HolidayBase):
...
 
+    def _populate(self, year):
         if year <= 1965:
             return None
 
         super()._populate(year)
 
-        _add_with_observed(
-            date(year, JAN, 1), "New Year's Day", "New Year's Day Holiday"
+        jan_1 = self._add_new_years_day("New Year's Day")
```
Comment (arkid15r) [inline holidays/countries/botswana.py:50]:
```
self._add_observed(
    self._add_new_years_day("New Year's Day")
)
```

for consistency?

File: holidays/countries/botswana.py [hunk 3]
```diff
@@ -30,58 +29,61 @@ class Botswana(HolidayBase):
...
 
-        _add_with_observed(date(year, JUL, 1), "Sir Seretse Khama Day")
+        self._add_observed(self._add_holiday("Sir Seretse Khama Day", JUL, 1))
 
-        # 3rd Monday of July = "President's Day"
-        dt = _get_nth_weekday_of_month(3, MON, JUL, year)
-
-        self[dt] = "President's Day"
-        self[dt + td(days=+1)] = "President's Day Holiday"
+        jul_3_mon = self._add_holiday(
```
Comment (arkid15r) [inline holidays/countries/botswana.py:69]: Not sure about the best naming option for this. I guess one of `third_monday_of_july`, `july_3rd_monday` would be better here (being different from jul_3 approach).

**Review by arkid15r — COMMENTED** [review]
🇧🇼 LGTM

File: holidays/countries/botswana.py (hunk 2 above)
Comment (KJhellico) [reply holidays/countries/botswana.py:50]: Here we need add observed for "New Year's Day" when "New Year's Day Holiday" already exist. When Jan 1 is Sunday, "New Year's Day (Observed)" should be Jan, 3 (Thuesday). And same for other 2-day holidays in Botswana.

Comment (arkid15r) [reply holidays/countries/botswana.py:50]: Would L82-85 syntax be acceptable here?

```
jan_1 = self._add_new_years_day("New Year's Day")
jan_2 = self._add_new_years_day_two("New Year's Day Holiday")
self._add_observed(jan_1)
self._add_observed(jan_2)
```

// sometimes observance is so tricky 🤯

**Pull request merged**